### PR TITLE
Allow for better customization of `sectionBy` macro.

### PR DESCRIPTION
**Pull request by mdrost**
I am not sure if order of parameters is good.

Summon @AlexVanderbist as macro creator.

### CODE REVIEW

**Comment by AlexVanderbist** [comment]
Thanks, this looks pretty good! I'll get back to you on the parameter order.

**Comment by mdrost** [comment]
I would like to mention that `sectionBy` has a flaw that cannot be fixed without breaking backward compatibility. The problem is that in output collection by default the section key is the same key that we are looking for in input collection. This causes a problem when we would like call `sectionBy('items')` as the keys will clash. It forces the user to use customization. IMHO by default the user should not be able to harm himself and both keys should has sensible default values.

**Comment by AlexVanderbist** [comment]
Good point, what would you propose as a sensible defaults? 

I like `items` for the items as it's also used in `groupByModel`. Would `section` work for the section key?

**Comment by mdrost** [comment]
In my opinion `section` is a good key.

**Comment by AlexVanderbist** [comment]
Tagged as `2.5.0`!

I'll add the breaking changes we discussed in the `laravel-55` branch.

**Comment by sebastiandedeyne** [comment]
I'm kind of a proponent of using no keys at all, just a two-element array (a tuple really). Then it could be chained with `eachSpread`, `mapSpread` etc, or can be destructured in PHP 7.1.

**Comment by mdrost** [comment]
Currently `int` keys are allowed, so default values could be `$sectionKey = 0, $itemsKey = 1` and if someone want to be fancy he could just call `sectionBy('derp', 'mySection', 'myItems')`.
I would like to point out that parameters order was not talked about. There is a quite lot of parameters including `$preserveKeys` parameter and that is the premise to using no keys at all.

File: src/macros.php [hunk 1]
```diff
@@ -193,26 +193,30 @@
...
+     * @param mixed $key
+     * @param mixed $sectionKey
+     * @param mixed $itemsKey
+     * @param bool $preserveKeys
      *
      * @return \Illuminate\Support\Collection
      */
-    Collection::macro('sectionBy', function ($sectionByKey) {
-        $sectionKeyRetriever = $this->valueRetriever($sectionByKey);
+    Collection::macro('sectionBy', function ($key, $sectionKey = null, $itemsKey = 'items', $preserveKeys = false): Collection {
```
Comment (mdrost) [inline src/macros.php]: @AlexVanderbist I just noticed that I forgot type hint `$preserveKeys` to `bool` :worried:

Comment (freekmurze) [inline src/macros.php]: @mdrost fyi You can just push more commits to your fork, they will show up in this PR.

**Pull request merged**